crates/tako/src/internal/server/core.rs

```diff
@@ -194,11 +194,13 @@ impl Core {
     pub fn new_worker(&mut self, worker: Worker) {
         /* Wake up sleeping tasks */
         let mut sleeping_sn_tasks = self.take_sleeping_tasks();
-        self.single_node_ready_to_assign
-            .append(&mut sleeping_sn_tasks);
-
+        if self.single_node_ready_to_assign.is_empty() {
```

Comment:
Is this always better? If `single_node_ready_to_assign` is long running (but often empty), it could accumulate capacity, while here we would reset the capacity every time.

Reply:
`single_node_ready_to_assign` is not accumulating capacity because scheduler calls `take_single_node_ready_to_assign` and consumes whole vec. I will look it, but the idea was to free potentially large vec that is not not needed for most of the lifetime of the server (usually only at the beginning).

Reply:
Ah, I didn't realize that scheduler takes the vec. Ok, then it should be fine.